pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -393,6 +398,27 @@ public void acknowledgeMessage(List<Position> positions, AckType ackType, Map<St
         }
     }
 
+    private void deleteTransactionMarker(PositionImpl position, AckType ackType, Map<String,Long> properties) {
+        if (position != null) {
+            ManagedLedgerImpl managedLedger = ((ManagedLedgerImpl) cursor.getManagedLedger());
+            managedLedger.asyncReadEntry(position, new ReadEntryCallback() {
+                @Override
+                public void readEntryComplete(Entry entry, Object ctx) {
+                    MessageMetadata messageMetadata = Commands.parseMessageMetadata(entry.getDataBuffer());
+                    if (Markers.isTxnCommitMarker(messageMetadata)) {
```

Comment:
I think we should delete all transaction markers right? not only the commit marker.